openedx/edx-ora2
# feat: add Google Cloud Storage support (GCS)

**TL;DR -** This is to add support to GCS files backend by adding a new option (`gcs`) to `ORA2_FILEUPLOAD_BACKEND`

JIRA: ---

**What changed?**

- `ORA2_FILEUPLOAD_BACKEND` environment variable accepts a new value `gcs` in addition to the existing options (`s3`, `filesystem`, `swift`, and `django`)
- No changes to the requirements. It is assumed that you already installed `django-storages[google]` instead of the default `django-storages` which will include `google-cloud-storages`

**Important to know!**
Using GSC with ORA means that `edx-platform` is already configured to use GCS instead of S3
- It is assumed that you already installed `django-storages[google]` instead of the default `django-storages` which will include `google-cloud-storages`
- It is assumed that `DEFAULT_FILE_STORAGE` is set to `storages.backends.gcloud.GoogleCloudStorage` in edx-platform. Other configurations might be needed too
- It is assumed that `edx-platform` functions directly and seamlessly with GCS without a proxy like MinIO or boto3

**Developer Checklist**

- [x] Translations and JS/SASS compiled~~ not needed

**Testing Instructions**

Using tutor development:
- Prepare your GSC storage, make sure to have a dedicated bucket for file upload
- Configure `edx-platform` to use GCS, including the installation of django-storages[google]
- Configure ORA2 to use GCS by setting `ORA2_FILEUPLOAD_BACKEND` to `gcs`
- Configure the dedicated bucket by setting `FILE_UPLOAD_STORAGE_BUCKET_NAME`
- Test uploading files from ORA2 and previewing them
- Test previewing uploaded files in ora2 MFE
- Verify that files exist in the dedicated storage
- Test deleting files
- Verify that files were deleted from the dedicated storage

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @openedx/content-aurora
